shared/actions/chat.js

```diff
@@ -238,6 +236,20 @@ function * _incomingMessage (action: IncomingMessage): SagaGenerator<any, any> {
         const conversationIDKey = conversationIDToKey(incomingMessage.convID)
         const message = _unboxedToMessage(messageUnboxed, 0, yourName, conversationIDKey)
 
+        // Is this message for the currently selected conversation? If so, mark
+        // it as read ASAP to avoid badging it -- we don't need to badge, the
+        // user's looking at it already.
+        const selectedSelector = (state: TypedState) => state.chat.get('selectedConversation')
+        const selectedConversationIDKey = yield select(selectedSelector)
+        if (message && message.messageID && conversationIDKey === selectedConversationIDKey) {
```

Comment:
@chrisnojima @gabriel Then in this scenario:

* app doesn't have focus

* we get a message

* you focus the app to read the message

How does the message get marked as read?  (Same question for Gabriel's red line last-message-read.)